### DISCUSSION TITLE
Interpolation usage docs

### DISCUSSION THREAD

**Discussion by xndyz** [pmndrs/react-spring#1637] [Support] [answered]
## 🤓 Question

It looks like `.interpolate` has been depreciated, but there are [currently no docs](https://react-spring.io/common/interpolation) on what the correct usage is. Currently the old API still works, but it seems odd that the new API would be pushed when the usage isn't clear.

Could someone provide a usage here and maybe we can look into adding that into the docs?

For example, how might one do this with the new API?

```
        <animated.div
          style={{
            opacity: styles.opacity,
            transform: styles.y.interpolate((y) => `translateY(${y}px)`),
          }}
        >
```

**Comment by joshuaellis** [comment, marked as answer]
It's definitely documented in the docs under [`interpolation`](https://react-spring.io/common/interpolation). It's simply renamed to `to` so your code would be come:

```js
<animated.div
  style={{
    opacity: styles.opacity,
    transform: styles.y.to((y) => `translateY(${y}px)`),
  }}
>
```

But you don't actually need to do that interpolation is `v9` of `react-spring` you could simply pass the `y` prop to the styles object 👍🏼

**Reply by xndyz** [reply]
Do you mean like this:

```
        <animated.div
          style={{
            opacity: styles.opacity,
            transform: `translateY(${styles.y}px)`,
          }}
        >
```

Because that doesn't work (but `to` does).

**Reply by joshuaellis** [reply]
no I mean like this:

```js
<animated.div
  style={{
    opacity: styles.opacity,
    y: styles.y,
  }}
>
```

**Reply by xndyz** [reply]
Ah I see, doing it that way is bad for performance. Looks like `to` is the right play here!
